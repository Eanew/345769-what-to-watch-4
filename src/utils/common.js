export const MINUTE_IN_SECONDS = 60;
export const HOUR_IN_MINUTES = 60;

export const Letter = {
  SPACE: ` `,
  COMMA: `,`,
  DASH: `-`,
};

export const Regular = {
  NUMBERS: /\d+/g,
  EXCEPT_NUMBERS: /(\D+)*[^.\d]/g,
  EMPTY_SPACE: /\s+/g,
  WORD_SEPARATION: /(\s|:\s|:)+/g,
  VALID_EMAIL: /^[a-zA-Z0-9.!#$%&â€™*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/,
};

export const Key = {
  SPACE: `Space`,
  ESC: `Escape`,
  ENTER: `Enter`,
};

export const isEscEvent = (evt, action) => {
  if (evt.key === Key.ESC) {
    action();
  }
};

export const toKebabCase = (defaultString) => defaultString
  .toLowerCase()
  .trim()
  .replace(Regular.WORD_SEPARATION, Letter.DASH);

export const toTwoDigit = (number) => number < 10 ? `0${number}` : number;

export const toTimeString = (seconds) => {
  const hours = Math.floor(seconds / MINUTE_IN_SECONDS / HOUR_IN_MINUTES) || null;

  const minutes = hours
    ? toTwoDigit(Math.floor(seconds / MINUTE_IN_SECONDS))
    : Math.floor(seconds / MINUTE_IN_SECONDS) || null;

  const restOfSeconds = minutes
    ? toTwoDigit(Math.floor(seconds % MINUTE_IN_SECONDS))
    : Math.floor(seconds % MINUTE_IN_SECONDS);

  return [hours, minutes, restOfSeconds].filter((timeUnit) => timeUnit !== null).join(`:`);
};

export const toDateTimeAttribute = (iso) => {
  const date = new Date(Date.parse(iso));

  return [date.getFullYear(), toTwoDigit(date.getMonth() + 1), toTwoDigit(date.getDate())].join(`-`);
};

export const sortByDate = (items, isoDateKey) => [...items]
  .sort((first, second) => Date.parse(second[isoDateKey]) - Date.parse(first[isoDateKey]));

export const toPercent = (max, current) => max ? Math.floor(current / max * 100) : 0;

export const extend = (a, b) => Object.assign({}, a, b);
