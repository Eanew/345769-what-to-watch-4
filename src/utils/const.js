export const Screen = {
  MAIN: `main`,
  DETAILS: `details`,
};

export const APPROVED_GENRES = [
  `Comedy`,
  `Crime`,
  `Documentary`,
  `Drama`,
  `Horror`,
  `Kids & Family`,
  `Romance`,
  `Sci-Fi`,
  `Thriller`];

export const Rating = {
  'Awesome': 10,
  'Very good': 8,
  'Good': 5,
  'Normal': 3,
  'Bad': 0,
};
