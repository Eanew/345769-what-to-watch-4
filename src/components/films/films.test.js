import React from "react";
import renderer from "react-test-renderer";
import Films from "./films.jsx";

import {toKebabCase} from "../../utils/common.js";
import {APPROVED_GENRES} from "../../utils/const.js";

it(`Should FilmsComponent render correctly`, () => {
  const filmsTitles = [
    `Fantastic Beasts: The Crimes of Grindelwald`,
    `Bohemian Rhapsody`,
    `Macbeth`];

  const tree = renderer.create(
      <Films
        films={filmsTitles.map((filmTitle, i) => ({
          id: filmTitle + i,
          filmTitle,
          release: 2011 + i,
          genre: APPROVED_GENRES[i],

          rating: {
            value: 3.4 + i,
            votesCount: 153 * i,
          },

          image: {
            preview: `img/${toKebabCase(filmTitle)}.jpg`,
            background: `img/bg-${toKebabCase(filmTitle)}.jpg`,
            poster: `img/${toKebabCase(filmTitle)}-poster.jpg`,
          },

          movie: {
            preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
            full: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
          },

          description: [
            `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave&rsquo;s friend and protege.`,

            `Gustave prides himself on providing first-class service to the hotel&rsquo;s guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave&rsquo;s lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],

          director: `Wes Andreson`,
          starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`][i],
        }))}
        onFilmCardClick={() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
