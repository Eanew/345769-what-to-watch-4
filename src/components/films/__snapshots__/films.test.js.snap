// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmsComponent render correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        controls={false}
        height="175"
        loop={true}
        muted={true}
        poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          type="video/mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        controls={false}
        height="175"
        loop={true}
        muted={true}
        poster="img/bohemian-rhapsody.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          type="video/mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        controls={false}
        height="175"
        loop={true}
        muted={true}
        poster="img/macbeth.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          type="video/mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
      >
        Macbeth
      </a>
    </h3>
  </article>
</div>
`;
